docs/user-guide/src/troubleshooting.md

```diff
@@ -88,3 +88,151 @@ of a forced deletion of its BareMetalHost object. If valid BMC credentials were
+The error will show in status and events.
+
+Example `kubectl get bmh -A`:
+
+```text
+NAMESPACE   NAME     STATE        CONSUMER   ONLINE   ERROR              AGE
+metal3      node-1   inspecting              true     inspection error   8m17s
+```
+
+BareMetalHost's events:
+
+```text
+Normal  InspectionError     5s     metal3-baremetal-controller  Failed to inspect hardware. Reason: unable to start inspection:
+Redfish exception occurred. Error: Setting boot mode to bios failed for node ceec28f5-cedb.rror: HTTP PATCH 
+https://192.168.111.1:8000/redfish/v1/Systems/... returned code 500.
+```
+
+## Provisioning errors
+
+Errors during provisioning will be visible when listing the BareMetalHosts:
+
+```text
+NAMESPACE   NAME     STATE          CONSUMER      ONLINE   ERROR                AGE
+metal3      node-1   provisioning   test1-dt8j2   true     provisioning error   149m
+```
+
+Check BareMetalHost's events for the specific reason.
+
+Wrong image checksum example:
+
+```text
+Normal  ProvisioningError   10m    metal3-baremetal-controller  Image provisioning failed: Deploy
+step deploy.write_image failed on node df880558-09da. Image failed to verify against checksum.
+location: CENTOS_9_NODE_IMAGE.img; image ID: /dev/sda; image checksum: abcd1234; verification checksum: ...
+```
+
+No root device found example:
+
+```text
+Normal  ProvisioningStarted  15s    metal3-baremetal-controller  Image provisioning started for http://172.22.0.1/images/CENTOS_9_NODE_IMAGE.img
+Normal  ProvisioningError    1s     metal3-baremetal-controller  Image provisioning failed: Deploy step deploy.write_image failed on node d25ce8de-914e-4146-a0c0-58825274572d. No suitable device was found for deployment using these hints {'name': 's== /dev/vdb'}
+```
+
+## No BareMetalHost available or matching
+
+This shows in the Metal3Machine status:
+
+```text
+Status:
+  Conditions:
+    Last Transition Time:  2025-08-15T10:53:05Z
+    Message:               No available host found. Requeuing.. Object will be requeued after 30s
+    Reason:                AssociateBMHFailed
+    Severity:              Error
+    Status:                False
+    Type:                  AssociateBMH
+```
+
+CAPM3 controller logs when there is no available hosts:
+
+```text
+I0815 11:10:35.699004   1 metal3machine_manager.go:332] "No available host found. Requeuing." logger="controllers.Metal3Machine.Metal3Machine-controller" metal3-machine="metal3/test-no-match-2" machine="test-no-match-2" cluster="test1" metal3-cluster="test1"
+```
+
+CAPM3 controller logs when the annotated host is not found:
+
+```text
+I0815 06:08:54.687380   1 metal3machine_manager.go:788] "Annotated host not found" logger="controllers.Metal3Machine.Metal3Machine-controller" metal3-machine="metal3/test1-zxzn7-qvl6n" machine="test1-zxzn7-qvl6n" cluster="test1" metal3-cluster="test1" host="metal3/node-0"
+```
+
+## Provider ID is missing
+
+Happens if `noCloudProvider` is set to `false` on Metal3Cluster when no external
```

Comment:
```suggestion
This occurs if `cloudProviderEnabled` is set to `true` on Metal3Cluster when no external cloud provider is used.
```
`noCloudProvider` is deprecated and will be removed, let's use the newer config `cloudProviderEnabled`